src/frequenz/sdk/actor/_channel_registry.py

```diff
@@ -49,3 +49,14 @@ def new_receiver(self, key: str) -> Receiver[Any]:
         if key not in self._channels:
             self._channels[key] = Broadcast(f"{self._name}-{key}")
         return self._channels[key].new_receiver()
+
+    async def _close_channel(self, key: str) -> None:
+        """Close a channel with the given key.
+
+        This method is private and should only be used in special cases.
+
+        Args:
+            key: A key to identify the channel.
+        """
+        if key in self._channels:
+            await self._channels[key].close()
```

Comment:
After closing, this should also remove the channel from the registry, from `self._channels`